tests/test_ess_csc.py

```diff
@@ -91,17 +91,16 @@ async def test_bin_script(self):
     async def validate_telemetry(self):
         for sensor_name in self.csc.device_configurations:
             device_configuration = self.csc.device_configurations[sensor_name]
-            telemetry_topic = getattr(
-                self.remote, f"tel_temperature{device_configuration.num_channels}Ch"
-            )
-            data = await telemetry_topic.next(flush=False)
+            data = await self.remote.tel_temperature.next(flush=False)
+            self.assertEquals(data.numChannels, device_configuration.num_channels)
```

Comment:
The next lines need update for temperature being an array. Please also check that the values past num_channels are all NaN (e.g. with math.isnan)

Reply:
I just pushed a code change that addresses exactly that.